### PR TITLE
Check existing property (fixes #11299)

### LINKED ISSUE #11299: Error creating a custom field since 4.4.0 [$100 awarded]
### Mautic Version

4.3.x series

### PHP version

8.0.20

### What browsers are you seeing the problem on?

Not relevant

### What happened?

Mautic 4.4.0 shows a validation error when creating a new custom field of type Date.

### How can we reproduce this issue?

Step 1: Do a clean install of Mautic 4.4.0
Step 2: Try to create a new custom field of type Date

image

### Relevant log output

_No response_

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<bountysource-plugin>

---
The **[$100 bounty](https://app.bountysource.com/issues/109790740-error-creating-a-custom-field-since-4-4-0?utm_campaign=plugin&utm_content=tracker%2F5355074&utm_medium=issues&utm_source=github)** on this issue has been claimed at [Bountysource](https://app.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F5355074&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by KN4CK3R**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [x]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #11299

#### Description:

Fixes #11299

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11321"></a>

### CODE REVIEW

**Comment by RCheesley** [comment]
@cla-bot check

Thanks for the PR @KN4CK3R and for signing the CLA!

Reviewers: See also #11320 for another potential fix.

**Review by RCheesley — DISMISSED** [review]
I have tested the PR with date, boolean and lookup fields which all seem to be working now. This will require code review to determine which is the most appropriate fix for the issue, and will need rebasing on the 4.4 branch if we are to accept in the next release. Looks like we may also need some code coverage improvement but that will be clarified when it is rebased.

File: app/bundles/LeadBundle/Form/Type/FieldType.php [hunk 1]
```diff
@@ -457,7 +457,7 @@ function (FormEvent $event) use ($formModifier, $disableDefaultValue) {
                     $data['defaultValue'] = null;
                 }
 
-                if (isset($data['type']) && !in_array($data['type'], $this->indexableFieldsWithLimits)) {
+                if (isset($data['type']) && isset($data['charLengthLimit']) && !in_array($data['type'], $this->indexableFieldsWithLimits)) {
                     $data['charLengthLimit'] = null;
                 }
```
Comment (escopecz) [inline app/bundles/LeadBundle/Form/Type/FieldType.php]: This was introduced in https://github.com/mautic/mautic/pull/11250. It should be part of another PR that is waiting to be pushed from our fork. The `charLengthLimit` field does not yet exist so I suggest to remove this if statement completely. WDYT?
```suggestion
```

Comment (KN4CK3R) [reply app/bundles/LeadBundle/Form/Type/FieldType.php]: Fine for me. Just wanted to keep the compatibility.

**Comment by escopecz** [comment]
Oh, right. Now there is unused property. Can you please remove it as well?

https://github.com/mautic/mautic/runs/7252491891?check_suite_focus=true

**Comment by KN4CK3R** [comment]
That's why I just checked if the field exists. Didn't want to remove other stuff.

**Review by RCheesley — DISMISSED** [review]
Re-testing the PR and the fields are able to be created, updated and deleted. Thanks @KN4CK3R ! 🚀

**Comment by RCheesley** [comment]
@all-contributors please add @KN4CK3R for code

**Comment by escopecz** [comment]
Looks great. One last request. Can you please rebase on 4.4 brach? That way we can merge it to the 4.4.1 bug fix release. There won't be any more feature releases for Mautic 4.

**Review by escopecz — DISMISSED** [review]
The code change makes sense. Thank you! 👍

**Comment by escopecz** [comment]
I added a functional test to ensure this won't happen again as it was a nasty regression bug.

**Review by escopecz — APPROVED** [review]
